### DISCUSSION TITLE
Change password with JS Client

### DISCUSSION THREAD

**Discussion by lukaszflorczak** [supabase/supabase#13506] [Questions] [answered]
I have doubts about how to properly and safely change a user password in an application (login with email + password). To clarify, I'm not referring to setting a new password when someone has forgotten their old one, but rather a situation where a user is logged in and wants to change their password in the settings.

I'm used to the solution where the old password should be provided in such a situation. Sometimes the same applies when changing an email. However, in Supabase this is done without providing the old password. Is this really safe?

I have read this thread: https://github.com/orgs/supabase/discussions/11412, but the proposed solutions don't seem good to me. Even if I implement my own solution to check the old correctness of the password, the method described here: https://supabase.com/docs/reference/javascript/v1/auth-update will still be available and possible to run if I use the JS SDK on client side, right?

**Comment by GaryAustin1** [comment]
I believe Supabase requires you have just signed in (not a refreshed session) in order to change the password.  Gotrue has this option GOTRUE_SECURITY_UPDATE_PASSWORD_REQUIRE_REAUTHENTICATION
There is no other option built in for verifying the password.

**Comment by mansueli** [comment, marked as answer]
This is now available in the dashboard:

<img width="866" alt="image" src="https://user-images.githubusercontent.com/5036432/233489316-b5efd121-3f5b-467b-a82d-120225fe2ded.png">

**Reply by lukaszflorczak** [reply]
Thank you for the answer. It’s great and I hope I’ll able to check and test it soon.
